tfhe/src/integer/gpu/list_compression/server_keys.rs

```diff
@@ -0,0 +1,239 @@
+}
+
+pub struct CudaDecompressionKey {
+    pub blind_rotate_key: CudaBootstrappingKey,
+    pub lwe_per_glwe: LweCiphertextCount,
+    pub parameters: PBSParameters,
+}
+
+impl CudaCompressionKey {
+    pub fn from_compression_key(compression_key: &CompressionKey, streams: &CudaStreams) -> Self {
+        Self {
+            packing_key_switching_key: CudaLwePackingKeyswitchKey::from_lwe_packing_keyswitch_key(
+                &compression_key.packing_key_switching_key,
+                streams,
+            ),
+            lwe_per_glwe: compression_key.lwe_per_glwe,
+            storage_log_modulus: compression_key.storage_log_modulus,
+        }
+    }
+
+    unsafe fn flatten_async(
+        vec_ciphertexts: &Vec<CudaRadixCiphertext>,
+        streams: &CudaStreams,
+    ) -> (CudaLweCiphertextList<u64>, Vec<CudaBlockInfo>) {
+        let first_ct = &vec_ciphertexts.first().unwrap().d_blocks;
+
+        // We assume all ciphertexts will have the same lwe dimension
+        let lwe_dimension = first_ct.lwe_dimension();
+        let ciphertext_modulus = first_ct.ciphertext_modulus();
+
+        // Compute total number of lwe ciphertexts we will be handling
+        let total_num_blocks: usize = vec_ciphertexts
+            .iter()
+            .map(|x| x.d_blocks.lwe_ciphertext_count().0)
+            .sum();
+
+        let lwe_ciphertext_count = LweCiphertextCount(total_num_blocks);
+
+        let gpu_index = streams.gpu_indexes[0];
+        let d_vec = unsafe {
+            let mut d_vec = CudaVec::new_async(
+                lwe_dimension.to_lwe_size().0 * lwe_ciphertext_count.0,
+                streams,
+                gpu_index,
+            );
+            let mut offset: usize = 0;
+            for ciphertext in vec_ciphertexts {
+                // Todo: We might use copy_self_range_gpu_to_gpu_async here
+                let dest_ptr = d_vec
+                    .as_mut_c_ptr(gpu_index)
+                    .add(offset * std::mem::size_of::<u64>());
+                let size = ciphertext.d_blocks.0.d_vec.len * std::mem::size_of::<u64>();
+                cuda_memcpy_async_gpu_to_gpu(
+                    dest_ptr,
+                    ciphertext.d_blocks.0.d_vec.as_c_ptr(gpu_index),
+                    size as u64,
+                    streams.ptr[gpu_index as usize],
+                    streams.gpu_indexes[gpu_index as usize],
+                );
+
+                offset += ciphertext.d_blocks.0.d_vec.len;
+            }
+
+            streams.synchronize();
+            d_vec
+        };
+
+        let flattened_ciphertexts =
+            CudaLweCiphertextList::from_cuda_vec(d_vec, lwe_ciphertext_count, ciphertext_modulus);
+
+        let info = vec_ciphertexts
+            .iter()
+            .flat_map(|x| x.info.blocks.clone())
+            .collect_vec();
+
+        (flattened_ciphertexts, info)
+    }
+
+    pub fn compress_ciphertexts_into_list(
+        &self,
+        ciphertexts: &Vec<CudaRadixCiphertext>,
```

Comment:
same can take &[CudaRadixCiphertext]

it's weird that clippy does not say anything, it's a very old/classic lint, we are not doing anything with the vec here